DictionarySlim trim

danmosemsft opened the pull request:
Relates to https://github.com/dotnet/corefxlab/issues/2592

2 commits - 
Remove Key/Value Enumerables to reduce instantiated size of type as discussed above
Remove Capacity to reduce unnecessarily publicness

danmosemsft commented:
Oh, let me add XML comments too

safern commented on tests/Microsoft.Experimental.Collections.Tests/DictionarySlimTests.cs:
```diff
@@ -723,6 +537,22 @@ public void UsedIEquatable2()
+            Object entries = fi.GetValue(dict);
+
+            PropertyInfo pi = typeof(Array).GetProperty("Length");
+            int capacity = (int)pi.GetValue(entries);
```
We could just return this?

danmosemsft commented:
OK now also 
Added XML docs (file temporarily included in the commit as I cannot find a way to consume $(OutputPath) in the project)
Removed Copy to (low value, extra size)

TylerBrinkley commented:
I think a key and value collection are still both needed just not the nested Enumerator struct. I think that's what @jkotas meant.

danmosemsft commented:
> I think a key and value collection are still both needed just not the nested Enumerator struct. I think that's what @jkotas meant.

I'm missing something, if we still have `ICollection<TKey>` but no Enumerator struct what would it return from `GetEnumerator<TKey>` ?

TylerBrinkley commented:
yield return? Though that would create an iterator class. I guess we can just use a LINQ Select clause to replicate the Key and Value collections where needed.

TylerBrinkley commented:
Looks good to me.

danmosemsft commented:
> I guess we can just use a LINQ Select clause to replicate the Key and Value collections where needed.

Yes, that's what I did in the tests. As with anything - it is always easier to add than remove API.

danmosemsft commented:
@safern is going to get the package signed and on NuGet in the next few days. After that is up @AnthonyLloyd are you interested in sending in the updated benchmark game submission? I think you deserve most of the credit.

AnthonyLloyd commented:
> @safern is going to get the package signed and on NuGet in the next few days. After that is up @AnthonyLloyd are you interested in sending in the updated benchmark game submission? I think you deserve most of the credit.

Yes sure, thanks you all. It may be a good idea to wait until the package has a few downloads. In the benchmarks game they want to see that it's not just a package put up for the benchmark only.

danmosemsft commented:
@anthonylloyd it already has 50K downloads :https://www.nuget.org/packages/Microsoft.Experimental.Collections/1.0.3-alpha
